packages/edit-site/src/components/add-new-template/utils.js

```diff
@@ -408,28 +418,96 @@ export const useTaxonomiesMenuItems = ( onClickMenuItem ) => {
 	return taxonomiesMenuItems;
 };
 
+export function useAuthorMenuItem( onClickMenuItem ) {
+	const existingTemplates = useExistingTemplates();
+	const defaultTemplateTypes = useDefaultTemplateTypes();
+	const authorInfo = useEntitiesInfo(
+		'root',
+		useAuthorMenuItem.AUTHOR_TEMPLATE_ENTITY_OBJECT
+	);
+	const authorMenuItem = defaultTemplateTypes?.find(
+		( { slug } ) => slug === 'author'
+	);
+	const hasGeneralTemplate = existingTemplates?.find(
+		( { slug } ) => slug === 'author'
+	);
+	if ( authorMenuItem && authorInfo.user?.hasEntities ) {
+		authorMenuItem.onClick = ( template ) => {
```

Comment:
This mutates the default template type. I assume that's not intentional. The other sources [take care to copy the object first](https://github.com/WordPress/gutenberg/blob/3584d7d50d08022ffe715b5f5050f69c5cb87299/packages/edit-site/src/components/add-new-template/utils.js#L156-L157).

```suggestion
	let authorMenuItem = defaultTemplateTypes?.find(
		( { slug } ) => slug === 'author'
	);
	const hasGeneralTemplate = existingTemplates?.find(
		( { slug } ) => slug === 'author'
	);
	if ( authorMenuItem && authorInfo.user?.hasEntities ) {
		authorMenuItem = { ...authorMenuItem };
		authorMenuItem.onClick = ( template ) => {
```